src/librustc/diagnostics.rs

```diff
@@ -1944,6 +1944,38 @@ you want. Example:
 ```
 "##,
 
+E0492: r##"
+A borrow of a constant containing interior mutability was attempted. Erroneous
+code example:
+
+```
+#![feature(const_fn)]
+
+use std::cell::Cell;
+
+const A: Cell<usize> = Cell::new(1);
+const B: &'static Cell<usize> = &A;
+// error: cannot borrow a constant which contains interior mutability, create
+//        a static instead
+
+// or:
+struct C { a: Cell<usize> }
+
+const D: C = C { a: Cell::new(1) };
+const E: &'static Cell<usize> = &D.a; // error
+
+// or:
+const F: &'static C = &D; // error
+
+fn main() {}
+```
+
+To solve this error, either use statics or avoid using `cell` type. If you want
```

Comment:
That only works for atomics, not for a cell holding a type of your choice. RefCell and Cell won't work because of UnsafeCell, and Mutex doesn't work because of destructors. So there's no safe way to place something like that in a static.